components/sync/service/brave_sync_service_impl.cc

```diff
@@ -349,31 +350,67 @@ void BraveSyncServiceImpl::OnSyncCycleCompleted(
 }
 
 void BraveSyncServiceImpl::UpdateP3AObjectsNumber() {
-  GetEntityCountsForDebugging(
+  GetEntityCountsForDebuggingBackport(
       BindRepeating(&BraveSyncServiceImpl::OnGotEntityCounts,
                     weak_ptr_factory_.GetWeakPtr()));
 }
 
 void BraveSyncServiceImpl::OnGotEntityCounts(
-    const syncer::TypeEntitiesCount& entity_count) {
-  // int total_entities = 0;
-  // for (const syncer::TypeEntitiesCount& count : entity_counts) {
-  //   total_entities += count.non_tombstone_entities;
-  // }
-
-  // if (GetUserSettings()->GetSelectedTypes().Has(
-  //         syncer::UserSelectableType::kHistory)) {
-  //   // History stores info about synced objects in a different way than the
-  //   // others types. Issue a separate request to achieve this info
-  //   sync_service_impl_delegate_->GetKnownToSyncHistoryCount(base::BindOnce(
-  //       [](int total_entities, std::pair<bool, int> known_to_sync_count) {
-  //         brave_sync::p3a::RecordSyncedObjectsCount(total_entities +
-  //                                                   known_to_sync_count.second);
-  //       },
-  //       total_entities));
-  // } else {
-  //   brave_sync::p3a::RecordSyncedObjectsCount(total_entities);
-  // }
+    const std::vector<TypeEntitiesCount>& entity_counts) {
+  int total_entities = 0;
+  for (const syncer::TypeEntitiesCount& count : entity_counts) {
+    total_entities += count.non_tombstone_entities;
+  }
+
+  if (GetUserSettings()->GetSelectedTypes().Has(
+          syncer::UserSelectableType::kHistory)) {
+    // History stores info about synced objects in a different way than the
+    // others types. Issue a separate request to achieve this info
+    sync_service_impl_delegate_->GetKnownToSyncHistoryCount(base::BindOnce(
+        [](int total_entities, std::pair<bool, int> known_to_sync_count) {
+          brave_sync::p3a::RecordSyncedObjectsCount(total_entities +
+                                                    known_to_sync_count.second);
+        },
+        total_entities));
+  } else {
+    brave_sync::p3a::RecordSyncedObjectsCount(total_entities);
+  }
+}
+
+void BraveSyncServiceImpl::GetEntityCountsForDebuggingBackport(
```

Comment:
Looks like concern from upstream is legit, if one of the callback in barrier failed, we won't get entity count at all in that case. Can we keep track of receiving count along with its type and record the count once only when we receive from all in the repeated callback? When the recording happens, we also clear the tracking state.

Reply:
so the main concern here is bound callback passed into `GetEntityCountsForDebuggingBackport` might not be called under some circumstances according to this [upstream change](https://chromium-review.googlesource.com/c/chromium/src/+/5388661)